tonic-build/src/server.rs

```diff
@@ -274,91 +287,187 @@ fn generate_trait_methods<T: Service>(
             quote!(&self)
         };
 
-        let method = match (
-            method.client_streaming(),
-            method.server_streaming(),
-            generate_default_stubs,
-        ) {
-            (false, false, true) => {
-                quote! {
-                    #method_doc
-                    async fn #name(#self_param, request: tonic::Request<#req_message>)
-                        -> std::result::Result<tonic::Response<#res_message>, tonic::Status> {
-                        Err(tonic::Status::unimplemented("Not yet implemented"))
+        #[cfg(not(feature = "async_trait"))]
```

Comment:
That's a lot of code duplication on top of what's already there. Given that the only difference is whether the method is prefixed with `async` or not and wrapping of the return type in `impl std::future::Future<Output = ..>`, can we simplify?

Reply:
Decomposing a method to such an extent may not be advisable. It represents a premature optimization that could lead to developer confusion and complications. Understanding the interdependencies among the components within the decomposition requires extra caution, potentially increasing the risk of errors.

Reply:
That's not a satisfying answer to me. I don't think the term "premature optimization" applies here, since there's no performance impact to what I'm suggesting (indeed, the performance might well be worse).

I'm suggesting that the increased code duplication makes this code harder to maintain (for example, carefully reviewing these changes is a pain in the ass), and asking you whether that issue could be mitigated.

Reply:
The changes have already been made. I was just expressing an opinion.

Reply:
@djc Gentle ping. Could you please take a look?